cmd/exporter.go

```diff
@@ -191,25 +71,28 @@ func Run(ctx *cli.Context) error {
 	onShutdown := make(chan os.Signal, 1)
 	signal.Notify(onShutdown, syscall.SIGINT, syscall.SIGTERM, syscall.SIGABRT)
 
-	stopPolling := make(chan bool)
-	c.pollProjects(stopPolling)
+	untilStopSignal := make(chan bool)
+	pipelinesOnInit := make(chan bool)
+	c.orchestratePolling(untilStopSignal, pipelinesOnInit)
+	// get immediately some data from the latest executed pipelines, if configured to do so
+	pipelinesOnInit <- cfg.OnInitFetchRefsFromPipelines
 
 	// Configure liveness and readiness probes
 	health := healthcheck.NewHandler()
-	if !cfg.Gitlab.SkipTLSVerify {
+	if !cfg.Gitlab.DisableHealthCheck {
```

Comment:
in the scenario where `DisableHealthCheck == false`, `DisableTLSVerify == true` and the `GitLab API does not have a valid certificate`, it will prevent it from succeeding the check

Reply:
does it? I was not able to test it... I might add one

Reply:
yes because `healthcheck.HTTPGetCheck(cfg.Gitlab.HealthURL, 5*time.Second)` will not validate it.

Reply:
got your point 👍 thanks, will fix it implementing `Check` in a func that will use the same transport as GitLab client

Reply:
this should be ok now

Reply:
really cool, thanks for the improvement!